client/src/constants/validation.js

```diff
@@ -28,6 +28,13 @@ const validateDateString = (s) => {
   return typeof date === 'number' && !Number.isNaN(date);
 };
 
+const validatePastDateString = (s) => {
+  if (/^\d{4}\/\d{2}\/\d{2}$/.test(s) === false) return false;
+  const date = Date.parse(s);
+  if (date > new Date()) return false;
+  return typeof date === 'number' && !Number.isNaN(date);
```

Comment:
Could skip some repeated logic here.
```suggestion
  if (!validateDateString(s)) return false;
  return Date.parse(s) <= new Date();
```